dds/DCPS/transport/rtps_udp/RtpsUdpTransport.cpp

```diff
@@ -305,11 +308,27 @@ RtpsUdpTransport::use_datalink(const RepoId& local_id,
   unsigned int blob_bytes_read;
   get_connection_addrs(remote_data, &uc_addrs, &mc_addrs, &requires_inline_qos, &blob_bytes_read);
 
+  NetworkAddress disco_addr_hint;
+  if (discovery_locator.length()) {
+    AddrSet disco_uc_addrs, disco_mc_addrs;
+    bool disco_requires_inline_qos;
+    unsigned int disco_blob_bytes_read;
+    get_connection_addrs(discovery_locator, &disco_uc_addrs, &disco_mc_addrs, &disco_requires_inline_qos, &disco_blob_bytes_read);
+
+    for (AddrSet::const_iterator it = disco_uc_addrs.begin(); it != disco_uc_addrs.end(); ++it) {
+      for (AddrSet::const_iterator it2 = uc_addrs.begin(); it2 != uc_addrs.end(); ++it2) {
+        if (it->addr_bytes_equal(*it2) && DCPS::is_more_local(disco_addr_hint, *it2)) {
+          disco_addr_hint = *it2;
```

Comment:
@jrw972 This is the loop. Currently, the discovery_locator blob should only have one address in it, but I left it open to accept multiple in case we ever wanted to pick from multiple.

Reply:
Thanks.